File: Data Structures & Algorithms/two-integer-sum-ii/submission-3.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        length = len(numbers)
        j =  length-1
        i = 0
        while(i!=j): 
            twoSum = numbers[i]+numbers[j]
            if(twoSum == target): 
                return [i+1, j+1]
            if(twoSum<target):
                i+=1
            elif(twoSum>target): 
                j-=1
        return []

File: Data Structures & Algorithms/two-integer-sum/submission-3.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int minValueNeeded = target;
        int[] achieveTarget = new int [nums.length];
        for(int i = 0; i < nums.length; i++)
            achieveTarget[i] = target;
        
        for(int i = 0; i < nums.length; i++){
            achieveTarget[i] -= nums[i];
            if(minValueNeeded>achieveTarget[i]){
                minValueNeeded = achieveTarget[i];
            }
            if(minValueNeeded<=nums[i]){
                for(int j = 0; j < i; j++){
                    if(achieveTarget[j]==nums[i])
                        return new int[]{j, i};
                }   
            }
        }
        return new int[]{};
    }
}
